### PR TITLE
Some visual improvements for the City Overview

### LINKED ISSUE #3839: Overview screen city table visual issues - alignment mainly
Compare these screenshots:
City-Overview-master
(taken with current master branch)
* The number columns are left aligned and thus do no longer align nicely with their header
* The second column looks cramped
* The first column's icon (white-circled settler) is right-aligned and therefore its function - sort by name - a little unclear. 

City-overview-aligned
(taken with a local modification)
* The numbers now align - using `cell.actor.setAlignment` instead of `cell.align`: I _don't_ understand why and why the current code worked not too long ago.
* Increased horizontal padding from 5f to 8f - teensy bit nicer I think.
* Using `getProductionForTileInfo` instead of `getCityProductionTextForCityButton` (orphaning it) looks IMHO better. I'd probably opt for a variant dropping cost to ensure 2-liners.
* `CityOverviewTable` line 43 has a call to left-align the settler icon - but no workaround I attempted did change the position. Stumped.

Opinions? Anyone who understands libGdx better than me know how to get that white icon back where it was intended to be?

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
A part of my changes to the Cities overview screen, resolves #3839
For some of my thoughts and screenshots see that issue.

* Alignment of columns
* Removed least interesting line of production info to ensure constant row height - sorry for the lazy comment, but gating the line with an optional bool would only make sense if the 3-line version will get used in the future...
* Push settler icon back to its intended left-aligned place
* Prepare for Faith: Increased column count will not break alignment or centering of title
* Moved code and commented blocks - I needed this to understand the code and my stashed code is based on this - main idea: keep stuff concerning one of the subtables as close together as possible.

**Not** part of this PR, stashed:
* Visual sort indicators
* Moved some but not all data that has essentially the same lifetime as the class instance to class level. Now a lot of stuff looks local to the constructor but survives by being part of nested Gdx structures and/or used in closures/delegates... My feeling is that the compiler and/or GC might have an easier time with such a rearrangement. Plus it allows forward references, so one could group data and helper functions a bit nicer.

### CODE REVIEW

File: core/src/com/unciv/ui/overviewscreen/CityOverviewTable.kt [hunk 1]
```diff
@@ -14,16 +17,22 @@ import kotlin.math.roundToInt
 
 class CityOverviewTable(val viewingPlayer: CivilizationInfo, val overviewScreen: EmpireOverviewScreen): Table() {
 
-    val columnsNames = arrayListOf("Population", "Food", "Gold", "Science", "Production", "Culture", "Happiness")
+    companion object {
```
Comment (yairm210) [inline core/src/com/unciv/ui/overviewscreen/CityOverviewTable.kt:20]: Why in a companion object though... what do you gain from this?

Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/overviewscreen/CityOverviewTable.kt:20]: Allows const keyword?

Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/overviewscreen/CityOverviewTable.kt:20]: I count roughly nine classes in Unciv that already have companion objects containing nothing but constants or constant-for-the-whole-app-run values. So where does in-code documentation to express "this `8` is a number of columns" go as best practice? I'd like to do this by declaring it as a constant with a descriptive name and somehow collect those for readability near the top of their file...

 I'll gladly help working towards a consistent best-practice coding style or documenting it...

Comment (yairm210) [reply core/src/com/unciv/ui/overviewscreen/CityOverviewTable.kt:20]: Const val isn't possible from within the class? O_O
Huh

**Comment by yairm210** [comment]
Oh
Uh
It seems I caused a conflict somewhere

**Comment by SomeTroglodyte** [comment]
As before, the 'resolve conflicts' page here has diff and prev/next but no UI for actual changes, accept theirs or anything... Your version of the conficting file should be 100% fine...

**Comment by SomeTroglodyte** [comment]
Rebasing this looked even scarier, but the result seems OK. Accepted your version of `CityConstructions.kt` without even going into the diff, and it _still_ added the missing EOL at the end of the file. Effect is, the lines again have changing heights, but now the middle line actually looks useful, so I'll look into equalizing vertical spacing by another method - minHeight should work.

**Pull request merged**